tests/test_backend.py

```diff
@@ -1726,6 +1742,26 @@ def test_isnan(self):
             self._run_test_case([_OUTPUT], {_INPUT: x_val})
             tf.reset_default_graph()
 
+    @staticmethod
+    def _has_specific_node(graph, checker):
```

Comment:
how about a ```group_nodes_by_type(g)```  returns a dict of node op type => list[node]?
then user may use the dict to do further validation. e.g. compare transpose count.
or for your case:
```group_nodes_by_type(g)["Upsample"].inputs[1]```

also, put it in tests/common.py, so other test scripts can use it.